src/pytorch_lightning/loggers/logger.py

```diff
@@ -190,12 +190,12 @@ def group_separator(self):
 
     @property
     @abstractmethod
-    def name(self) -> str:
+    def name(self) -> Optional[str]:
         """Return the experiment name."""
 
     @property
     @abstractmethod
-    def version(self) -> Union[int, str]:
+    def version(self) -> Optional[Union[int, str]]:
```

Comment:
I don't think these should be Optional

Reply:
The current implementation of the `WandbLogger` does return `None` if name is `None` and `experiment` does not exist. I can change the default for both version and name to be `""` instead of `None` and remove the Optional.

Reply:
Resolving in favor of keping the current logic. Thanks